packages/medulas-react-components/src/context/ToastProvider/index.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { Toast } from './Toast';
+
+export enum ToastVariant {
+  SUCCESS = 'success',
+  WARNING = 'warning',
+  ERROR = 'error',
+  INFO = 'info',
+}
+
+const SharedToast = (): JSX.Element => (
+  <ToastConsumer>
+    {({ open, message, onClose, variant }: ToastContextInterface) => {
+      return (
+        <Toast
+          open={open}
+          onClose={onClose}
+          message={message}
+          variant={variant}
+        />
+      );
+    }}
+  </ToastConsumer>
+);
+
+export interface ToastContextInterface {
+  readonly showToast: (message: string, variant: ToastVariant) => void;
+  readonly onClose: () => void;
+  readonly open: boolean;
+  readonly message: string;
+  readonly variant: ToastVariant;
+}
+
+export const ToastContext = React.createContext<ToastContextInterface>({
+  showToast: (message: string, variant: ToastVariant) => ({ message, variant }),
+  onClose: () => null,
+  open: false,
+  message: '',
+  variant: ToastVariant.INFO,
+});
+
+interface Props {
+  readonly children: React.ReactNode;
+}
+
+interface ToastState {
+  readonly open: boolean;
+  readonly message: string;
+  readonly variant: ToastVariant;
+}
+
+export const ToastProvider = ({ children }: Props): JSX.Element => {
+  const [toast, setToast] = React.useState<ToastState>({
+    open: false,
+    message: '',
+    variant: ToastVariant.INFO,
+  });
+
+  const showToast = (message: string, variant: ToastVariant): void =>
+    setToast({
+      message,
+      variant,
+      open: true,
+    });
+
+  const closeToast = (): void =>
+    setToast({
+      message: '',
+      open: false,
+      variant: ToastVariant.INFO,
+    });
+
+  const toastConfig = {
```

Comment:
The only information which needs to be passed down to tree is the `show` and `close` functionality, in this case, when a component does:
```
const { show, close} = useContext(ToastContext);
```
why do we need to expose the rest of messages?

Reply:
Yes, but SharedToast component (ToastProvider/index.tsx  line: 11) required some of this information.